passes/opt/opt_rmdff.cc

```diff
@@ -320,6 +338,16 @@ bool handle_dff(RTLIL::Module *mod, RTLIL::Cell *dff)
 		val_init.bits.push_back(bit.wire == NULL ? bit.data : RTLIL::State::Sx);
 	}
 
+	if (sig_e.size()) {
+		if (!sig_e.is_fully_const())
+			return false;
+		if (sig_e != val_ep) {
+			if (has_init)
+				mod->connect(sig_q, val_init);
```

Comment:
It should even set the output to `val_init` if `has_init` is false. Otherwise one might get warnings for undriven used signals.